cbmgmtx/errors.go

```diff
@@ -58,3 +58,16 @@ func (e ServerInvalidArgError) Unwrap() error {
 func (e ServerInvalidArgError) Error() string {
 	return fmt.Sprintf("%s: %s - %s", e.Unwrap().Error(), e.Argument, e.Reason)
 }
+
+type ServerNotFoundError struct {
```

Comment:
I would have made individual BucketNotFoundError, CollectionNotFoundError, etc... types with fields that match the specific type (for instance, a BucketName field for BucketNotFoundError).  I'm not really convinced that my approach is better though.  What was the reasoning behind your approach?

Reply:
In hindsight, maybe I'm actually convinced your approach is along the lines of a better one.  Perhaps considering how Golang "stacks" error types via it's unwrap mechanism, maybe this "ServerNotFoundError" should actually be something like a "ResourceError", and thus can be reused for the "AlreadyExists" style (and others) errors as well?

Reply:
My thinking was that we just wanted to maintain the same behaviour, with the ability to access the resource path if need be. Having one struct that wraps all the not found errors achieves this while declaring as few new structs as possible since we can just use errors.As to distinguish between the resource types and then just read the path. In this line of thinking I agree that we should make it more generic, then we can use if for other error types.

Reply:
Changes this to ResourceError